cms/envs/common.py

```diff
@@ -1115,3 +1120,25 @@
     'options': {},
 }
 PROCTORING_SETTINGS = {}
+
+
+############################ OAUTH2 Provider ###################################
+
+# OpenID Connect issuer ID. Normally the URL of the authentication endpoint.
+OAUTH_OIDC_ISSUER = 'https://www.example.com/oauth2'
+
+# OpenID Connect claim handlers
+OAUTH_OIDC_ID_TOKEN_HANDLERS = (
+    'oauth2_provider.oidc.handlers.BasicIDTokenHandler',
+    'oauth2_provider.oidc.handlers.ProfileHandler',
+    'oauth2_provider.oidc.handlers.EmailHandler',
+    # Includes the "administrator" claim in the token. This currently resides in lms/djangoapps!
+    'oauth2_handler.IDTokenHandler'
```

Comment:
The 1.8 upgrade was a red herring.

ID tokens created as part of the OIDC flow used by our IDAs include the "administrator" claim by way of the [`EXTRA_SCOPES` setting](https://github.com/edx/cookiecutter-django-ida/blob/master/%7B%7Bcookiecutter.repo_name%7D%7D/%7B%7Bcookiecutter.repo_name%7D%7D/settings/base.py#L167-L168) consumed by [edx/auth-backends](https://github.com/edx/auth-backends/blob/9e0e2ef43ba92751051724b55641d50d5a9538c3/auth_backends/backends.py#L20). Signing a staff user into the Programs service via OIDC forces the creation of a token with the "administrator" claim. The scopes requested by Programs appear to be retained by the user's access token. When Studio later requests an ID token for the user, the user's existing access token is used, the "permissions" scope is detected, and the "administrator" claim is included in the token, allowing unpublished programs to be listed:

![](https://i.imgur.com/bl8gYYB.png)

I wouldn't have caught this without destroying my devstack, forcing me to sign into Programs again and create a new ID token. The tricky part remains finding a good way to specify the "permissions" scope for tokens requested in process (instead of by an IDA using OIDC).